### PR TITLE
Preftech metadata on app registration

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes https://github.com/spring-cloud/spring-cloud-dataflow/issues/1214

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java [hunk 1]
```diff
@@ -191,9 +203,32 @@ else if (!CollectionUtils.isEmpty(apps)) {
...
+	 * Trigger early resolution of the metadata resource of registrations that have an explicit metadata artifact.
+	 * This assumes usage of {@link org.springframework.cloud.deployer.resource.support.DelegatingResourceLoader}.
+	 */
+	private void prefetchMetadata(List<AppRegistration> appRegistrations) {
+		forkJoinPool.execute(() -> {
+			appRegistrations.stream()
+				.filter(r -> r.getMetadataUri() != null)
+				.parallel()
+				.forEach(r -> {
+					System.out.println("Getting " + r.getMetadataUri());
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java]: This needs to be removed.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java [hunk 2]
```diff
@@ -191,9 +203,32 @@ else if (!CollectionUtils.isEmpty(apps)) {
...
 	}
 
+	/**
+	 * Trigger early resolution of the metadata resource of registrations that have an explicit metadata artifact.
+	 * This assumes usage of {@link org.springframework.cloud.deployer.resource.support.DelegatingResourceLoader}.
+	 */
+	private void prefetchMetadata(List<AppRegistration> appRegistrations) {
+		forkJoinPool.execute(() -> {
+			appRegistrations.stream()
+				.filter(r -> r.getMetadataUri() != null)
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java]: Does this mean during the prefetch, we don't download the artifact where the app itself contains the metadata resource? ie) only prefetch if there are explicit metadata resource available for the app registration?

Comment (ericbottard) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/AppRegistryController.java]: That's exactly that

**Comment by ilayaperumalg** [comment]
LGTM, merging.

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as 058de3a

**Pull request closed without merging**